Services/UI/OJS.Services.Ui.Data/Implementations/ContestsDataService.cs

```diff
@@ -210,17 +210,20 @@ public Task<bool> IsUserInExamGroupByContestAndUser(int id, string userId)
             {
                 case 0:
                     return contests
-                        .OrderByDescending(c => c.EndTime);
+                        .OrderBy(c => c.EndTime.HasValue && c.EndTime.Value < DateTime.UtcNow ? 2 :
+                            c.EndTime.HasValue ? 1 : 3)
+                        .ThenBy(c => c.EndTime);
```

Comment:
This seems too complex. Will this approach order them how you want?
```
.OrderByDescending(c => c.CanBeCompeted)
.ThenBy(c => c.EndTime);
```

Reply:
My initial approach was to sort them either by the IsActive property or CanBeCompeted but I am getting the following error: 

{
    "title": "An unexpected error occurred!",
    "status": 500,
    "detail": "The LINQ expression 'DbSet<Contest>()\r\n    .Where(c => !(c.IsDeleted))\r\n    .Where(c => c.IsVisible)\r\n    .OrderByDescending(c => c.CanBeCompeted)' could not be translated. Additional information: Translation of member 'CanBeCompeted' on entity type 'Contest' failed. This commonly occurs when the specified member is unmapped. Either rewrite the query in a form that can be translated, or switch to client evaluation explicitly by inserting a call to 'AsEnumerable', 'AsAsyncEnumerable', 'ToList', or 'ToListAsync'. See https://go.microsoft.com/fwlink/?linkid=2101038 for more information.",
    "instance": "63680756-46b4-40d1-83c6-dd2c771f1953",
    "extensions": {}
}

Properties on the Contest model are marked with the NotMapped attribute and it doesn't allow me to use them in the query. This is the work around i came up with since it sorts them into active, past, not containing endtime and them sorts them by the end date.